src/Microsoft.ML.AutoML/AutoMLExperiment/TrialSettingsProposer/HyperParameterProposer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.ML.SearchSpace;
+
+namespace Microsoft.ML.AutoML
+{
+    internal class HyperParameterProposer : ITrialSettingsProposer
+    {
+        private readonly Dictionary<string, ITuner> _tuners;
+        private readonly IServiceProvider _provider;
+
+        public HyperParameterProposer(IServiceProvider provider)
+        {
+            this._tuners = new Dictionary<string, ITuner>();
+            this._provider = provider;
```

Comment:
You do this in other files, but is there a reason you are using `this` instead of just leaving it out?

Reply:
You mean using `this` as prefix for private fields/method? It's more of a personal preference because sometimes it's hard to tell if a variable is a temporary variable or not simply depends on `_`.

Reply:
Our coding standards/guide wants us to only use `this` if it is absolutely necessary. [https://github.com/dotnet/machinelearning/blob/f6111c3ed7a1fe147aa9bac5dfa03f875cd6e51e/.editorconfig#L48-L49](https://github.com/dotnet/machinelearning/blob/f6111c3ed7a1fe147aa9bac5dfa03f875cd6e51e/.editorconfig#L48-L49). You can remove `this` except when it really is needed?

Reply:
Can I do this on a separate PR? `Microsoft.ML.SearchSpace` also use `this` prefix and I'd like to fix them altogether.

Reply:
Yeah, that sounds good to me.